blocks/library/paragraph/index.js

```diff
@@ -93,9 +92,55 @@ registerBlockType( 'core/paragraph', {
 		}
 	},
 
+	inspector( { attributes, setAttributes } ) {
```

Comment:
A little bit concerned about this. Some times we need local state to be shared beween the inspector, toolbar and the `edit`. If we separate this, it's not possible anymore.

Do you think it's possible to add some magic to `InspectorControls` to detect if it's a multiselection and only render for one block only or something?

Reply:
I tried it before, but can't remember it well, so I'll try it again. Could you give an example of local state, just for me to think about solutions for it? :)

Reply:
We don't have an exemple with shared local state between the `edit` and `BlockInspector` but we do have between `edit` and `BlockControls` (in the `audio` or `block` blocks) and I believe these two are similar because in the exact same way we can display the same block toolbar in we multi-select similar blocks.

Reply:
Exactly, what is done here for the inspector should then also be done for the toolbar. The problem with the edit function is that it makes use for `focus`. We'd need to remove that and figure out how to do that differently. I'll find the example with shared state and think about it.

Reply:
@youknowriad Which block are you referring to with the local state? Is there anything that wouldn't be solvable with block attributes? Ideally, we should stick with those for state.

Reply:
@iseulde block attributes are "state" but they are "state" persisted in post_content or comments or meta. I don't know if it's a good idea to use them for "local state", we may end up serializing useless local state attributes in comments.

Reply:
For example, the `audio` block triggers an `editing` local state using a button toolbar. And this local state changes the UI of the block.

Reply:
I'm looking at this for the block toolbar (so if the selected blocks are of the same type, they get the toolbar and the controls apply changes to all selected blocks.)

It seems to me that having an `inspector` and `toolbar` function like this is very close to what we need, but as we need to share state between edit and BlockInspector/Toolbar, we need to reuse those functions to fill the appropriate slots.

The render in edit would look like this:

```
<BlockControls>
    { this.toolbar( { state, attributes, setAttributes } ) }
</BlockControls>
<InspectorControls>
    { this.inspector( { state, attributes, setAttributes } ) }
</InspectorControls> 
```

The slots would get filled as normal, single block selection behaviour in unchanged.

Then, for example, in the block inspector where we currently have "Coming Soon", we'd get the block settings for the type of blocks we have selected, and call `settings.inspector` passing an appropriate state, attributes, and a setAttributes function that would set attributes across all selected blocks. The state and attributes could be reduced from the selected blocks, so that if all blocks had the same value for a control, it would get that value (for example, if all paragraphs were the same color, that color would be selected).